### PR TITLE
[RFR] Avoid overiding all players when cleaning all guesses

### PR DESCRIPTION
**Pull request by JulienMattiussi**

### CODE REVIEW

File: src/sudoku.js [hunk 1]
```diff
@@ -171,64 +171,65 @@ export const getPuzzleLeaderboard = (puzzle) => {
...
-    if (
-      indexCol != col &&
-      puzzle[indexRow][col].value &&
-      puzzle[indexRow][col].value === value &&
-      !puzzle[indexRow][col].guess
-    ) {
-      return true;
+    for (let row = 0; row < 9; row++) {
+        if (
+            indexRow != row &&
```
Comment (Nitix) [inline src/sudoku.js:176]: It can be useful to create a function for this.

**Review by Nitix — CHANGES_REQUESTED** [review]
Can you provides some tests too?

File: src/sudoku.js (hunk 1 above)
Comment (JulienMattiussi) [reply src/sudoku.js:176]: out of scope

Comment (JulienMattiussi) [reply src/sudoku.js:176]: but you can open a PR

**Comment by JulienMattiussi** [comment]
matthieu ? out of this body

**Pull request merged**